src/main/java/com/example/demo/validator/FacilityValidator.java

```diff
@@ -0,0 +1,48 @@
+package com.example.demo.validator;
+
+import com.example.demo.exception.facility.FacilityAlreadyExistsException;
+import com.example.demo.exception.facility.FacilityIllegalArgumentException;
+import com.example.demo.model.facility.Facility;
+import com.example.demo.model.facility.FullFacilityDataDTO;
+import com.example.demo.repository.FacilityRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+import java.util.Objects;
+
+@Component
+@RequiredArgsConstructor
+public class FacilityValidator {
+    private final FacilityRepository facilityRepository;
+    private final DoctorValidator doctorValidator;
+
+    public void validateFacilityCreation(FullFacilityDataDTO facilityData) {
+        validateFacilityData(facilityData);
+        validateNameAvailability(facilityData.name());
+        doctorValidator.validateDoctorBulkCreation(facilityData.doctors());
+    }
+
+    public void validateFacilityEdit(Facility facility, FullFacilityDataDTO facilityData) {
+        validateFacilityData(facilityData);
+        if (facilityRepository.existsByName(facilityData.name()) && !Objects.equals(facility.getName(), facilityData.name())) {
+            throw new FacilityAlreadyExistsException("Facility with name: %s already exists.".formatted(facilityData.name()), OffsetDateTime.now());
+        }
+    }
+
+    private void validateFacilityData(FullFacilityDataDTO facilityData) {
+        if (facilityData.name() == null
```

Comment:
to jest normalne natomiast czasem mozecie sobie uzyc Objects.isNull(facilityData.name() )